internal/telemetry/event.go

```diff
@@ -141,11 +143,32 @@ func withDuration(cmd *cobra.Command) eventOpt {
 	}
 }
 
+func withCI() EventOpt {
+	return func(event Event) {
+		_, ok := os.LookupEnv("CI")
+		event.Properties["ci"] = ok
+	}
+}
+
+func withAnonymousID() EventOpt {
+	return func(event Event) {
+		id, err := machineid.ProtectedID(config.ToolName)
+		_, _ = log.Debugf("error generating machine id: %v\n", err)
+		event.Properties["anonymous_id"] = id
```

Comment:
can we have a fallback mechanism here in case of an error?
Like generating a GUID and saving it alongside config.toml (e.g. ~/Library/Application Support/atlascli/aid)

Reply:
I've created a ticket https://jira.mongodb.org/browse/CLOUDP-201093